scm-webapp/src/main/java/sonia/scm/api/v2/resources/ResourceLinks.java

```diff
@@ -27,17 +27,18 @@
 import sonia.scm.security.gpg.UserPublicKeyResource;
 
 import javax.inject.Inject;
+import javax.inject.Provider;
 import java.net.URI;
 import java.net.URISyntaxException;
 
 @SuppressWarnings("squid:S1192")
 // string literals should not be duplicated
 class ResourceLinks {
 
-  private final ScmPathInfoStore scmPathInfoStore;
+  private final Provider<ScmPathInfoStore> scmPathInfoStore;
```

Comment:
This whole change is unnecessary. You should use a `Provider` where the resource links are injected.